pyccel/codegen/printing/ccode.py

```diff
@@ -430,14 +430,14 @@ def _print_LiteralComplex(self, expr):
             return self._print(PyccelAssociativeParenthesis(PyccelMul(expr.imag, LiteralImaginaryUnit())))
         else:
             return self._print(PyccelAssociativeParenthesis(PyccelAdd(expr.real,
-                            PyccelMul(expr.imag, LiteralImaginaryUnit()))))
+                            PyccelMul(expr.imag, LiteralImaginaryUnit()), simplify = True)))
```

Comment:
If you do not simplify `PyccelMul` then this `simplify` will have no effect. However if you added a `simplify` to `PyccelMul` to generate the appropriate `LiteralComplex` then the expression would correctly simplify to a `LiteralComplex`. But (big but), this is **dangerous**. If you simplify here, you could cause an infinite loop. We are trying to print a `LiteralComplex`. In order to do that in c we need to print:
```
real + imag * _Complex_I
```
If you simplify here then you will get a `LiteralComplex` which is printed by `_print_LiteralComplex` which needs to print at this line which simplifies to a `LiteralComplex` which is printed by `_print_LiteralComplex` etc